src/Workspaces/Core/Desktop/Workspace/FileTextLoader.cs

```diff
@@ -198,5 +211,18 @@ private string GetDebuggerDisplay()
         {
             return nameof(Path) + " = " + Path;
         }
+
+        private static void ValidateFileLength(Workspace workspace, string path)
+        {
+            var fileLength = FileUtilities.GetFileLength(path);
+            if (fileLength > workspace.Options.GetOption(FileTextLoaderOptions.FileLengthThreshold))
+            {
+                // log max file length to AI
+                Logger.Log(FunctionId.FileTextLoader_FileLengthThresholdExceeded, KeyValueLogMessage.Create(m => m["FileLength"] = fileLength));
+
+                var message = string.Format(WorkspacesResources.File_size_exceeds_maximum_allowed_size_0, path);
+                throw new IOException(message);
```

Comment:
+1 I don't think we should be throwing an exception here. I'd either change it to Try-pattern, or define a new exception type  that's gonna be handled here: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/TextDocumentState.cs,177